cmd/client.go

```diff
@@ -55,10 +57,32 @@ func (cli *Client) RunNode(c *clipkg.Context) error {
 	defer app.Stop()
 	logNodeBalance(store)
 	logConfigVariables(config)
+	logIfNonceOutOfSync(store)
 
 	return cli.errorOut(cli.Runner.Run(app))
 }
 
+func logIfNonceOutOfSync(store *strpkg.Store) {
+	add := store.TxManager.GetActiveAccount().Address
+
+	var transactions []models.Tx
+	query := store.Select(q.Eq("From", add))
+	if err := query.Limit(1).OrderBy("Nonce").Reverse().Find(&transactions); err == storm.ErrNotFound {
+		return
+	}
```

Comment:
Let's place lines 68-72 into its own method in `store/models/orm.go`. `orm.GetLastNonce(address common.Address) (uint64, error)` or something.

And since we're querying by certain fields, let's add an index for them so we're performant:
```
// Tx contains fields necessary for an Ethereum transaction with
// an additional field for the TxAttempt.
type Tx struct {
	ID       uint64         `storm:"id,increment,index"`
	From     common.Address `storm:"index"` // new index
	To       common.Address
	Data     []byte
	Nonce    uint64 `storm:"index"` // new index
	Value    *big.Int
	GasLimit uint64
	TxAttempt
}
```
https://github.com/smartcontractkit/chainlink/blob/master/store/models/eth.go#L17

Regarding the index `From     common.Address storm:"index"`, the bolt/storm db handles common.Address differently to string. Make sure we're consistent in how we save and query.